### PR TITLE
HARP-4670: Host docs and demo in s3

### PR DESCRIPTION
**Pull request by musculman**
Host docs and demo in s3 following a structure so that there is an archive of releases, and an index file of those releases maintained.
Remove extra script from travis and add it to typedoc generation.
New minisite under /docs with assets that show the available releases.
This minisite is copied as well by typedoc and hosted on s3.
For every release, the minisite is pushed to gh-pages and the docs and examples to s3.
For every master merge, the master documentation on s3 is updated.
This PR depends on #429

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,8 +38,8 @@ jobs:
         set -ex
         yarn run build
         yarn run typedoc
-        echo -e 'include:\n  - "_*"' > dist/_config.yml
       deploy:
+        before_deploy: ./scripts/prepare_doc_deploy.sh
```
Comment (keinkonzept) [inline .travis.yml]: Shouldn't this line go one level up, same level as `deploy` itself?

File: docs/css/style.css [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
...KCQk8cGF0aCBpZD0iaGVyZSIgY2xhc3M9InN0MiIgZD0iTTM4LjIsNDAuNWMtMi4zLTIuNy0yLjItNC4yLTAuOC01LjZjMS43LTEuNywzLjUtMSw1LjUsMC45TDM4LjIsNDAuNXogTTYxLjgsMTAuNQoJCQljMS43LTEuNywzLjUtMSw1LjUsMC45bC00LjcsNC43QzYwLjIsMTMuNCw2MC4zLDEyLDYxLjgsMTAuNXogTTc2LDE1LjNjLTIuMSwzLjItNS45LDguNC05LjgsNC41bDkuOS05LjlDNzUuMiw5LDc0LjYsOC4yLDc0LDcuNwoJCQljLTUuMy01LjMtMTEuMy01LjQtMTYtMC43Yy0zLjEsMy4xLTQsNi43LTMuMSwxMC4xbC0zLjEtMy41QzUxLDE0LDQ3LjIsMTcuNCw1MCwyMi40bC0zLjUtMi45bC00LjcsNC43bDYuNCw2LjQKCQkJYy00LjktMy45LTEwLjItMy41LTE0LjQsMC43Yy00LjUsNC41LTQuMiwxMC0wLjgsMTQuNWwtMC43LTAuN2MtNC41LTQuNS05LjQtMi45LTExLjgtMC41Yy0xLjksMS45LTMsNC40LTIuNSw2LjJsLTkuNy05LjcKCQkJbC01LjIsNS4ybDE5LjIsMTkuMmgxMC4zbC02LjktNi45Yy0zLjYtMy43LTMuNy01LjYtMS45LTcuNGMxLjctMS43LDMuNy0wLjYsNy4yLDIuOGw2LjgsNi44bDUuMS01LjFsLTYuNS02LjUKCQkJYzQuNywzLjUsMTAuMywzLjcsMTUuMy0xLjNjMCwwLDAuMS0wLjEsMC4xLTAuMWwwLDBjMy4xLTIuOCw0LTUuNiw0LTUuNmwtMy45LTIuNmMtMi4xLDMuMi01LjgsOC40LTkuNyw0LjVsOS45LTkuOWw2LjIsNi4yCgkJCWw1LjQtNS40bC03LjctNy43Yy0zLjYtMy42LTEuNS03LDAtOC4zYzAuNywxLjUsMS43LDIuOSwzLDQuMmM1LDUsMTEuNCw2LDE3LjEsMC40YzAsMCwwLjEtMC4xLDAuMS0wLjFsMCwwYzMuMS0yLjgsNC01LjYsNC01LjYKCQkJTDc2LDE1LjN6Ii8+Cgk8L2c+CjwvZz4KPGcgaWQ9IkViZW5lXzIiPgo8L2c+Cjwvc3ZnPg==');
```
Comment (keinkonzept) [inline docs/css/style.css]: Storing background image in-line here and loading another image from file later on line 42 does not look consistent. Why not using the same approach everywhere?

File: docs/index.html [hunk 3]
```diff
@@ -1,9 +1,77 @@
-<!DOCTYPE HTML>
-<html lang="en-US">
-    <head>
-        <meta charset="UTF-8">
-        <meta http-equiv="refresh" content="0; url=./doc">
-        <title>Page Redirection</title>
-    </head>
-    <body></body>
+<!doctype html>
```
Comment (zbigg) [inline docs/index.html]: Rest of html in project is indented with 4 spaces, please do so here too.

File: docs/index.html [hunk 4]
```diff
@@ -1,9 +1,77 @@
...
+<html lang="en">
+	<head>
+		<meta charset="utf-8">
+		<title>harp.gl</title>
+		<meta name="description" content="Landing page for HERE Maps on Github">
```
Comment (keinkonzept) [inline docs/index.html]: Should it be more specific and mentioning harp.gl?

File: scripts/prepare_doc_deploy.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+if [ "$TRAVIS_BRANCH" != "master" ]; then
+    print_date=`date +"%d-%m-%y"`
+    new_release='{ \"date\": \"'${print_date}'\",\"hash\": \"'${folder_name}'\"}'
+    wget 'http://harp.gl.s3-website-us-east-1.amazonaws.com/docs/releases.json'
+
+    if [ ! -f releases.json ]; then
+        echo -e '[\n'${new_release}'\n]\n' > releases.json
+    else
+        sed -i -e "2i$new_release,\n" releases.json >> releases.json
```
Comment (zbigg) [inline scripts/prepare_doc_deploy.sh]: `sed -i` already changes file in place, so either remove -i or don't redirect it (again?) to same file

File: docs/index.html [hunk 6]
```diff
@@ -1,9 +1,77 @@
...
+				<span class="imprint">
+						<span class="copyright_string label_block">&copy; 2019	HERE</span> |
+						<a target="_blank" href="https://legal.here.com/en/terms/serviceterms/us/" class="label_block">Terms</a> |
+						<a target="_blank" href="https://legal.here.com/en/privacy/policy/us/" class="label_block">Privacy &amp;
+								Cookies</a>
+				</span>
+		</footer>
+
+		<script type="text/javascript">
+			let dropdown = $('#releases-dropdown');
```
Comment (keinkonzept) [inline docs/index.html]: This is not guaranteed to work until entire page is loaded. Please, use something like [this](https://learn.jquery.com/using-jquery-core/document-ready/).

File: docs/index.html [hunk 7]
```diff
@@ -1,9 +1,77 @@
...
-<html lang="en-US">
+<!doctype html>
+<html lang="en">
     <head>
-        <meta charset="UTF-8">
-        <meta http-equiv="refresh" content="0; url=./doc">
-        <title>Page Redirection</title>
+        <meta charset="utf-8">
+        <title>harp.gl</title>
+        <meta name="description" content="Landing page for HERE Maps on Github">
```
Comment (CraigBerlin) [inline docs/index.html]: "GitHub" is official name

**Comment by musculman** [comment]
Successfully tested in this fork: https://github.com/musculman/harp.gl

**Comment by musculman** [comment]
Live demo can be seen here: https://musculman.github.io/harp.gl/

**Pull request merged**